resources/dw/resource_hive_vw_acc_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2024 Cloudera. All Rights Reserved.
+//
+// This file is licensed under the Apache License Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, either express or implied. Refer to the License for the specific
+// permissions and limitations governing your use of the file.
+
+package dw_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/client/operations"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/models"
+	"github.com/cloudera/terraform-provider-cdp/cdpacctest"
+	"github.com/cloudera/terraform-provider-cdp/utils"
+)
+
+type hiveTestParameters struct {
+	Name              string
+	ClusterID         string
+	DatabaseCatalogID string
+}
+
+func HivePreCheck(t *testing.T) {
+	errMsg := "AWS Terraform acceptance testing requires environment variable %s to be set"
+	if _, ok := os.LookupEnv("CDW_CLUSTER_ID"); !ok {
+		t.Fatalf(errMsg, "CDW_CLUSTER_ID")
+	}
+	if _, ok := os.LookupEnv("CDW_DATABASE_CATALOG_ID"); !ok {
+		t.Fatalf(errMsg, "CDW_DATABASE_CATALOG_ID")
+	}
+}
+
+func TestAccHive_basic(t *testing.T) {
+	params := hiveTestParameters{
+		Name:              cdpacctest.RandomShortWithPrefix(cdpacctest.ResourcePrefix),
+		ClusterID:         os.Getenv("CDW_CLUSTER_ID"),
+		DatabaseCatalogID: os.Getenv("CDW_DATABASE_CATALOG_ID"),
+	}
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			cdpacctest.PreCheck(t)
+			HivePreCheck(t)
+		},
+		ProtoV6ProviderFactories: cdpacctest.TestAccProtoV6ProviderFactories,
+		CheckDestroy:             testCheckHiveDestroy,
+		Steps: []resource.TestStep{
+			// Create and Read testing
+			{
+				Config: utils.Concat(
+					cdpacctest.TestAccCdpProviderConfig(),
+					testAccHiveBasicConfig(params)),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("cdp_vw_hive.test_hive", "name", params.Name),
+					resource.TestCheckResourceAttr("cdp_vw_hive.test_hive", "cluster_id", params.ClusterID),
+					resource.TestCheckResourceAttr("cdp_vw_hive.test_hive", "database_catalog_id", params.DatabaseCatalogID),
+				),
+			},
+			// TODO ImportState testing
+			// TODO Update and Read testing
```

Comment:
Do you know if these are going to be implemented?

Reply:
Maybe not in the first iteration, because we discussed we should focus on the create and delete of the objects first. However I think we should implement it in a long term.